libs/experimental/langchain_experimental/llms/ollama_functions.py

```diff
@@ -93,17 +98,12 @@ class _AllReturnType(TypedDict):
 def parse_response(message: BaseMessage) -> str:
     """Extract `function_call` from `AIMessage`."""
     if isinstance(message, AIMessage):
-        kwargs = message.additional_kwargs
-        if "function_call" in kwargs:
-            if "arguments" in kwargs["function_call"]:
-                return kwargs["function_call"]["arguments"]
-            raise ValueError(
-                f"`arguments` missing from `function_call` within AIMessage: {message}"
-            )
-        raise ValueError(
-            "`function_call` missing from `additional_kwargs` "
-            f"within AIMessage: {message}"
-        )
+        tool_calls = message.tool_calls
+        if len(tool_calls) > 0:
+            tool_call = tool_calls[-1]
+            args = tool_call.get("args")
+            return json.dumps(args)
```

Comment:
Looks good. Consider retaining the check for additional_kwargs as a fallback (e.g., `elif "function_call" in kwargs`).

Reply:
added code for backwards compatibility.